Draft agency API spec

adborden opened the pull request:
Resolves https://github.com/18F/foia/issues/32

Thanks @bsweger for drafting the bulk of this and consulting on the final touches. Once merged, this is ready for a first round of feedback from agencies.

Commit:
Add a draft API request sample

adborden committed:
Request parameters

- Add request parameters
- Tweak the formatting

adborden committed:
Use table for parameters

adborden committed:
Note on authentication

adborden committed:
Rename spec

bsweger committed:
A few wording tweaks for consideration (#67)

adborden committed:
Feedback from @bsweger

adborden committed:
Add section for the 10% fields

bsweger reviewed (approved):
Thanks for the updates, @adborden. You're right to address the "10%" in this spec, and I really like the GSA example you added.

Left a few minor questions below, but my one larger piece of feedback is...I can't shake this feeling that we're making an artificial distinction between the types of FOIA questions/answers being passed to the API.
 We talk internally a lot about the 90% and the 10%,
 but is that meaningful to the recipient of the request?

One idea, if you want to maintain this distinction, might be to consider a different name for `agency_form_fields`. That's throwing me off a bit...for example, `agency` and `agency_component_name` are not `agency_form_fields`, but `contract_number` is. 

Maybe something like `component_specific_fields` or `agency_specific_fields`? To make it clearer that these aren't form fields and they aren't about the agency; rather it's info needed to process a FOIA request for this particular agency component.

Am marking this as approved because even if we don't resolve these questions, this is in good shape to shop around and get feedback on.

hbillings reviewed (approved):
This looks good to me! Clear, helpful examples and language.

adborden committed:
rfc

- Rename `agency_form_fields` which is confusing as all of the fields are agency
  form fields
- Clarify that these are agency component specific fields
- Remove formal field definitions for sample agency component fields to avoid
  confusion
- Clarify sentence about portal validation for agency component fields
- Include missing GSA.json
- Add agency component fields to the curl example

adborden committed:
Fix heading

We are referring to agency components here, not agencies.
